src/decoder.js

```diff
@@ -28,6 +28,15 @@ function decode({ json, complete }, token) {
     // Parse payload if needed
     if (json === true || header.typ === 'JWT') {
       payload = JSON.parse(payload)
+
+      // https://tools.ietf.org/html/rfc7519#section-7.2
+      //
+      // 10.  Verify that the resulting octet sequence is a UTF-8-encoded
+      //      representation of a completely valid JSON object conforming to
+      //      RFC 7159 [RFC7159]; let the JWT Claims Set be this JSON object.
+      if (typeof payload !== 'object') {
```

Comment:
typeof null (which is valid top level json as well) will pass this

Reply:
Added the check, good spot!